File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/server/messages/ServerMessageRemoveClient.java
package net.stzups.board.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.server.websocket.Client;
import net.stzups.board.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageRemoveClient extends ServerMessage {
    Client client;

    public ServerMessageRemoveClient(Client client) {
        super(ServerMessageType.REMOVE_CLIENT);
        this.client = client;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeShort(client.getId());
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageRemoveClient.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageRemoveClient extends ServerMessage {
    id: number;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.id = byteBuffer.readInt16();
    }

    getType(): ServerMessageType {
        return ServerMessageType.REMOVE_CLIENT;
    }
}